pkg/reconciler/namespace/namespace.go

```diff
@@ -236,24 +254,34 @@ func (c *Controller) ensureScheduled(ctx context.Context, lclusterName string, n
 			return err
 		}
 		if !isValidCluster {
-			// Currently assigned cluster no longer exists or is not ready.
+			// The currently assigned cluster no longer exists or is not ready.
 			if err := c.assignCluster(ctx, ns); err != nil {
 				return err
 			}
 		}
 	}
+
 	return nil
 }
 
 // ensureScheduledStatus ensures the status of the given namespace reflects whether the
 // namespace has been assigned to a cluster.
 func (c *Controller) ensureScheduledStatus(ctx context.Context, ns *corev1.Namespace) error {
-	scheduled := (ns.Labels[clusterLabel] != "")
-	statusChangeRequired := scheduled != IsScheduled(ns)
-	if statusChangeRequired {
-		return c.updateSchedulableCondition(ctx, ns, scheduled)
+	scheduled := ns.Labels[clusterLabel] != ""
+	disabled := !scheduleRequirement.Matches(labels.Set(ns.Labels))
+
+	if condition := conditions.Get(&NamespaceConditionsAdapter{ns}, NamespaceScheduled); condition != nil {
+		conditionStatus := scheduled != (condition.Status == corev1.ConditionTrue)
```

Comment:
nit: `s/conditionStatus/conditionStatusChanged/` ? i.e. indicate clearly in the name what the conditional indicates. Same comment for `conditionReason` below

Reply:
Agreed. I've updated both variables as per your suggestion.